### PR TITLE
DRILL-8386: Add Support for User Translation for Cassandra

### PR DESCRIPTION
**Pull request by cgivre**
# [DRILL-8386](https://issues.apache.org/jira/browse/DRILL-8386): Add Support for User Translation for Cassandra

## Description
Adds support for user translation for Apache Cassandra.

## Documentation
Updated README.

## Testing
Added unit tests.

### CODE REVIEW

File: contrib/storage-splunk/README.md [hunk 1]
```diff
@@ -42,6 +42,10 @@ Sometimes Splunk has issue in connection to it:
 https://github.com/splunk/splunk-sdk-java/issues/62 <br>
 To bypass it by Drill please specify "reconnectRetries": 3. It allows you to retry the connection several times.
 
+### User Translation
+The Splunk plugin supports user translation.  Simply set the `authMode` parameter to `USER_TRANSLATION` and use either the plain or vault credential provider for credentials.
```
Comment (jnturton) [inline contrib/storage-splunk/README.md:46]: Do you think I should make authMode values case insensitive before 1.21 is released? USER_TRANSLATION looks odd compared to other Drill config values and is only in caps due to the coincidence that the Java Enum names are in caps.

File: contrib/storage-cassandra/src/test/java/org/apache/drill/exec/store/cassandra/CassandraUserTranslationTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    String sql = "SHOW DATABASES WHERE schema_name LIKE '%cassandra%'";
+
+    RowSet results = client.queryBuilder().sql(sql).rowSet();
+    assertEquals(1, results.rowCount());
+    results.clear();
+  }
+
+  @Test
+  public void testInfoSchemaQueryWithValidCredentials() throws Exception {
+    // This test validates that the cassandra connection with user translation appears whne the user is
```
Comment (jnturton) [inline contrib/storage-cassandra/src/test/java/org/apache/drill/exec/store/cassandra/CassandraUserTranslationTest.java:58]:
```suggestion
    // This test validates that the cassandra connection with user translation appears when the user is
```

**Review by jnturton — APPROVED** [review]
I'm not trying to take over the requested review job here but I've been over it and can leave a +1.

File: contrib/storage-splunk/README.md (hunk 1 above)
Comment (cgivre) [reply contrib/storage-splunk/README.md:46]: That's probably a good idea.  I'd give it a +1.

**Comment by cgivre** [comment]
@jnturton  Thanks for the review!

**Pull request merged**